### PR TITLE
[BIOMAGE-2071] - Install Datadog as sidecar

### PR DESCRIPTION
**Pull request by aerlaut**
# Description

# Details
#### URL to issue
N/A
#### Link to staging deployment URL (or set N/A)
N/A

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: chart-infra/templates/datadog-sidecar-permissions.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (kafkasl) [inline chart-infra/templates/datadog-sidecar-permissions.yaml:1]: same issue as in the pipeline PR

**Review by kafkasl — CHANGES_REQUESTED** [review]

**Review by kafkasl — APPROVED** [review]

File: chart-infra/templates/datadog-sidecar-permissions.yaml (hunk 1 above)
Comment (aerlaut) [reply chart-infra/templates/datadog-sidecar-permissions.yaml:2]: See replies and discussion in the comment: https://github.com/biomage-org/pipeline/pull/30/files#r984552629

File: chart-infra/templates/pod-template.tpl [hunk 2]
```diff
@@ -49,6 +49,36 @@
...
+            value: '{"*": "%%label%%"}'
+          # Disable log collection by DD agent
+          # because we push logs to Cloudwatch
+          - name: DD_LOGS_ENABLED
+            value: "false"
+          - name: DD_CONTAINER_EXCLUDE
+            value: "name:.*"
+          - name: DD_CONTAINER_INCLUDE_METRICS
+            value: "name:worker name:worker-r"
+          - name: DD_KUBERNETES_KUBELET_NODENAME
```
Comment (ivababukova) [inline chart-infra/templates/pod-template.tpl:76]: this is aligned differently from here: https://github.com/biomage-org/pipeline/pull/30/files 
Which one of the two is correct?

File: chart-infra/templates/worker-spawn-right-to-api.yaml [hunk 3]
```diff
@@ -6,9 +6,9 @@ metadata:
...
 rules:
-- apiGroups: [""]
-  resources: ["pods"]
-  verbs: ["get", "list", "patch"]
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "list", "patch"]
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
```
Comment (ivababukova) [inline chart-infra/templates/worker-spawn-right-to-api.yaml:14]: same here -- remove changes. Unless you are sure and have tested that they work

**Review by ivababukova — APPROVED** [review]

**Pull request merged**